server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -254,8 +239,57 @@ public void putSynonymsSet(String resourceName, SynonymRule[] synonymsSet, Actio
         }));
     }
 
+    public void putSynonymRule(String synonymsSetId, SynonymRule synonymRule, ActionListener<UpdateSynonymsResult> listener) {
+        checkSynonymSetExists(synonymsSetId, listener.delegateFailure((l1, obj) -> {
```

Comment:
Why do we need this `checkSynonymSetExists`?   Can we just assume a user want to create new synonymsSet with this new SynonymRule? This adds an extra search request, which I think is not necessary.

Reply:
I'd say that could be confusing - in case a user mistypes the synonyms set ID, then instead of an error, she would have a new synonyms set.

I think we should err on the side of caution, and make the user aware that the synonyms set does not exist and needs to be created.

Reply:
Synchronized with @carlosdelest offline, and we decided to keep the current implementation with a search request.  But added a TODO item for GA to deal with empty collections.